inlong-sort/sort-flink/base/src/main/java/org/apache/inlong/sort/base/sink/TableChange.java

```diff
@@ -177,9 +177,80 @@ public String toString() {
 
     final class DeleteColumn implements ColumnChange {
 
+        private final String[] fieldsNames;
+
+        public DeleteColumn(String[] fieldsNames) {
+            Preconditions.checkArgument(fieldsNames.length > 0, "Invalid filed name: at least one is required");
+            this.fieldsNames = fieldsNames;
+        }
+
         @Override
         public String[] fieldNames() {
-            return new String[0];
+            return fieldsNames;
+        }
+
+        @Override
+        public boolean equals(Object o) {
+            if (this == o) {
+                return true;
+            }
+            if (!(o instanceof DeleteColumn)) {
+                return false;
+            }
+            DeleteColumn that = (DeleteColumn) o;
+            return Arrays.equals(fieldsNames, that.fieldsNames);
+        }
+
+        @Override
+        public int hashCode() {
+            return Arrays.hashCode(fieldsNames);
+        }
+
+        @Override
+        public String toString() {
+            return String.format("DELETE COLUMNS `%s`", fieldsNames[fieldsNames.length - 1]);
+        }
+    }
+
+    final class UpdateColumn implements ColumnChange {
+
+        private final String[] fieldsNames;
+        private final LogicalType dataType;
+        private final boolean isNullable;
+        private final String comment;
+
+        public UpdateColumn(String[] fieldsNames, LogicalType dataType, boolean isNullable, String comment) {
+            Preconditions.checkArgument(fieldsNames.length > 0, "Invalid filed name: at least one is required");
```

Comment:
at least one field

Reply:
it can not distinguished (add + delete) or modify, example first [a, b, c] -> delete c [a, b] -> add d [a, b, d], it is judged as unknown change, and I will judeg it by using information extracted from DDL in next version.